data/dataloader.py

```diff
@@ -0,0 +1,48 @@
+from type import (
+    PreprocessConfig,
+    DatasetCategories,
+    TrainDataset,
+    TestDataset,
+)
+from typing import List, Callable, Tuple, Optional, Union
+from .transformation import transform_dataset
+from datasets.load import load_dataset
+
+from .merge import merge_datasets
+
+
+class DataLoaderBase:
```

Comment:
why do we need a class and not just pass in functions?

Reply:
We are defining preprocess_config here and then overwriting it later. So with this pattern there is unfortunately a stateful situation.

Reply:
I can live with it as a class, just curious

Reply:
yeah, I see